buildingmotif/dataclasses/shape_collection.py

```diff
@@ -321,6 +321,30 @@ def gensym():
     ]
     clauses += " UNION ".join(tc_clauses)
 
+    # handle targetSubjectsOf
+    targetSubjectsOf = graph.objects(shape, SH.targetSubjectsOf)
+    tso_clauses = [
+        f"?target {tso.n3()} ?ignore .\n" for tso in targetSubjectsOf  # type: ignore
+    ]
+    clauses += " UNION ".join(tso_clauses)
+
+    # handle targetObjectsOf
+    targetObjectsOf = graph.objects(shape, SH.targetObjectsOf)
+    too_clauses = [
+        f"?ignore {too.n3()} ?target .\n" for too in targetObjectsOf  # type: ignore
+    ]
+    clauses += " UNION ".join(too_clauses)
+
+    # handle targetNode
+    targetNode = graph.objects(shape, SH.targetNode)
+    if len(targetNode) > 1:
```

Comment:
This appears to be the line where the test is failing. I think you may need to `next` the generator and catch the error as you can't check the length of a generator.